typescript/src/codegen.ts

```diff
@@ -1,196 +1,177 @@
-    },
-  ],
-};
-
-async function generateTs(spec: CompilationSpec): Promise<void> {
-  const out = `${spec.outputPath}/${spec.outputType}/${spec.contractName}`;
-  const name = spec.contractName;
-
-  const schemas = readSchemas({
-    schemaDir: spec.schemaDir,
-    argv: { packed: false },
-  });
-  return await generate(name, schemas, out);
-}
-
-function getSchemaDirectories(
-  rootDir: string,
-  contracts?: string
-): Promise<string[][]> {
-  return new Promise((resolve, reject) => {
-    const contractList = contracts?.split(",").map((dir) => dir.trim()) ?? [];
-    const directories: string[][] = [];
-    if (contractList.length) {
-      // get the schema directory for each contract
-      for (const contractName of contractList) {
-        const schemaDir = path.join(rootDir, contractName, "schema");
-        directories.push([schemaDir, contractName]);
-      }
-      resolve(directories);
-    } else {
-      // get all the schema directories in all the contract directories
-      fs.readdir(rootDir, (err, dirEntries) => {
-        if (err) {
-          console.error(err);
-          return;
-        }
-        if (!dirEntries) {
-          console.warn(`no entries found in ${rootDir}`);
-          resolve([]);
-          return;
-        }
-        dirEntries.forEach((entry) => {
-          try {
-            const schemaDir = path.resolve(rootDir, entry, "schema");
-            if (
-              fs.existsSync(schemaDir) &&
-              fs.lstatSync(schemaDir).isDirectory()
-            ) {
-              directories.push([schemaDir.replaceAll("\\", "/"), entry]);
-            } else {
-              log(`${schemaDir} is not a directory`, LogLevels.Verbose);
-            }
-          } catch (e) {
-            console.warn(e);
-          }
-        });
-        resolve(directories);
-      });
-    }
-  });
-}
-
-async function main() {
-  let config = {
-    ...DEFAULT_CONFIG,
-  };
-
-  const compilationSpecs: CompilationSpec[] = [];
-  log("Calculating generation specs...");
-  for (const root of config.schemaRoots) {
-    const { name, paths, outputName, outputDir } = root;
-    for (const path of paths) {
-      const schemaDirectories = await getSchemaDirectories(path);
-      for (const [directory, contractName] of schemaDirectories) {
-        compilationSpecs.push({
-          contractName: contractName,
-          schemaDir: directory,
-          outputPath: outputDir,
-          outputType: outputName,
-        });
-      }
-    }
-  }
-  log(`code generating for ${compilationSpecs?.length ?? 0} specs...`);
-  if (CODEGEN_LOG_LEVEL === LogLevels.Debug) {
-    console.log("Compilation specs:");
-    console.dir(compilationSpecs);
-  }
-
-  const codegenResponses: Promise<void>[] = [];
-  for (const spec of compilationSpecs) {
-    codegenResponses.push(generateTs(spec));
-  }
-  await Promise.all(codegenResponses);
-
-  log(`code generation complete`, LogLevels.Normal);
-}
-
-main();
```

Comment:
any particular reason to remove this? this new one breaks if you add or remove a contract atm which is a bit of a bummer.

Reply:
the script had to change to deal with the different output schemas / the new codegen, but we can prob find a way to not have to specify contract names specifically

Reply:
no worries, I made the script a bit better